src/reporter.cc

```diff
@@ -144,20 +144,30 @@ void MetadataReporter::SendMetadata(
 
   std::vector<json::value> entries;
   for (auto& entry : metadata) {
-    const MonitoredResource& resource = entry.first;
+    const std::string& full_resource_name = entry.first;
     MetadataStore::Metadata& metadata = entry.second;
     if (metadata.ignore) {
       continue;
     }
     json::value metadata_entry =
-        json::object({  // MonitoredResourceMetadata
```

Comment:
Is there a name for the structure you're sending here? If so, can the new code reference that name?

Reply:
This is the ResourceMetadata structure in the publish request. Added.